ddtrace/propagation/http.py

```diff
@@ -483,10 +489,201 @@ def _extract(headers):
+        values that would be in x-datadog-tags as well as those needed for propagation information.
+        The keys to the ``dd`` values have been shortened as follows to save space:
+        ``sampling_priority`` = ``s``
+        ``origin`` = ``o``
+        ``_dd.p.`` prefix = ``t.``
+    """
+
+    @staticmethod
+    def _is_valid_datadog_trace_tag_key(key):
+        return key.startswith("t.")
+
+    @staticmethod
+    def _get_traceparent_values(headers):
+        # type: (Dict[str, str]) -> Optional[Tuple[str,int,int,int]]
+
+        tp = _extract_header_value(_POSSIBLE_HTTP_HEADER_TRACEPARENT, headers)
+        if tp is None:
+            return None
+
+        try:
+            version, trace_id_hex, span_id_hex, trace_flags_hex = tp.split("-")
+            # check version is a valid hexadecimal
+            int(version, 16)
+            # currently 00 is the only version format, but if future versions come up we may need to add changes
+            if version != "00":
+                log.warning("unsupported traceparent version:%s . Will still attempt to parse.", (version))
+
+            if len(trace_id_hex) == 32 and len(span_id_hex) == 16 and len(trace_flags_hex) >= 2:
+                trace_id = _hex_id_to_dd_id(trace_id_hex)
+                span_id = _hex_id_to_dd_id(span_id_hex)
+
+                # All 0s are invalid values
+                assert trace_id != 0
+                assert span_id != 0
+
+                trace_flags = _hex_id_to_dd_id(trace_flags_hex)
+                # there's currently only one trace flag, which denotes sampling priority
+                # was set to keep "01" or drop "00"
+                # trace flags is a bit field: https://www.w3.org/TR/trace-context/#trace-flags
+                sampling_priority = int(trace_flags & 0x1)
+
+            else:
+                log.debug("W3C traceparent hex length incorrect: %s", tp)
+                raise ValueError
+
+        except (ValueError, AssertionError):
+            log.exception("received invalid w3c traceparent: %s.", tp)
+            return None
+
+        return tp, trace_id, span_id, sampling_priority
+
+    @staticmethod
+    def _get_tracestate_values(ts):
+        # type: (str) -> Optional[Tuple[Optional[int], Dict[str, str], Optional[str]]]
+
+        # tracestate parsing, example: dd=s:2;o:rum;t.dm:-4;t.usr.id:baz64,congo=t61rcWkgMzE
+        # store ts so we keep other vendor data
+        meta = {_TRACESTATE_KEY: ts}
+
+        try:
+            result = re.search("dd=(.+?),", ts)
+
+            # grab dd value if dd is only list member
+            if not result:
+                result = re.search("dd=(.*)", ts)
+            try:
+                dd = dict(item.split(":") for item in result.group(1).split(";"))  # type: ignore
+            except AttributeError:
+                log.debug(("no dd list member in tracestate from incoming request: %r "), ts)
+                dd = None
```

Comment:
This simplifies the regex and removes a redundant try/except block. If `dd = dict(....)` fails we will log: `received invalid dd header value in tracestate: [ts value]`. 

Would something like this work? 

```suggestion
           # match datadog tracestate (excludes trailing comma)
            dd_ts = re.search("dd=(.+?)(?:,|$)", ts)
            dd = None
            if dd_ts:
                dd = dict(item.split(":") for item in dd_ts.split(";"))  # type: ignore
```

Reply:
ah great regex thanks! I just used `dd = dict(item.split(":") for item in dd_ts.group(1).split(";"))` instead and it looks good. Thanks!